### PR TITLE
CI enhancement - step1 make tfvars can be recorded

### PR DESCRIPTION
**Pull request by xueli181114**
What I did in the PR:
- Make cluster creation record the tfvars
- Updated README.md
- Moved `Create` to `Apply` to avoid confusion
- Changed part of attributes to use pointer
- Exclude some cases may unstable, will do enhancement in next PR
image
@amalykhi @shellymiron @yuwang-RH @nirarg could you please review?

### CODE REVIEW

**Comment by yuwang-RH** [comment]
@xueli181114 Two comments are left, others /lgtm

File: tests/e2e/machine_pool_test.go [hunk 1]
```diff
@@ -747,111 +745,115 @@ var _ = Describe("TF Test, default machinepool day-2 testing", func() {
...
-				Expect(err).NotTo(HaveOccurred())
-				defaultMachinepoolArgFromMachinepool, err := exe.BuildDefaultMachinePoolArgsFromDefaultMachinePoolState(defaultMachinePoolResource)
-				Expect(err).NotTo(HaveOccurred())
-				defaultMachinepoolArgFromMachinepool.Cluster = clusterID
-				defaultMachinepoolArgFromMachinepool.Token = token
-				dmpArgFromMachinepoolForTesting := defaultMachinepoolArgFromMachinepool
-
-				By("Edit the taints without additional machinepool")
+			It("Author:yuwan-High-OCP-69009 Check the default machinepool creation with the cluster and edit/delete it via terraform",
+				ci.Day2, ci.Critical, ci.FeatureMachinepool, ci.Exclude,
```
[the remark this answers is not in the document]
Comment (xueli181114) [reply tests/e2e/machine_pool_test.go:761]: @yuwang-RH This case caused defaut machinepool deletion. I excluded it from this PR and will enable it after we get it fixed.

File: tests/e2e/machine_pool_test.go [hunk 2]
```diff
@@ -747,111 +745,115 @@ var _ = Describe("TF Test, default machinepool day-2 testing", func() {
...
-				output, err = dmpService.Destroy()
-				Expect(err).ToNot(HaveOccurred())
-				Expect(output).To(ContainSubstring("Warning: Cannot delete machine pool"))
+				// Only check this when confirm no other machinepool existing
+				if resp.Total() == 1 {
+					output, err = dmpService.Destroy()
+					Expect(err).ToNot(HaveOccurred())
+					Expect(output).To(ContainSubstring("Warning: Cannot delete machine pool"))
+				}
+
```
[the remark this answers is not in the document]
Comment (xueli181114) [reply tests/e2e/machine_pool_test.go:755]: Reasonable, I will recover it back.

[the remark this answers is not in the document]
Comment (xueli181114) [reply tests/e2e/machine_pool_test.go:754]: Recovered

**Pull request merged**